### DISCUSSION TITLE
LESSON 11 - No Metadata?

### DISCUSSION THREAD

**Discussion by Programmusic** [smartcontractkit/full-blockchain-solidity-course-py#147] [Q&A] [answered]
Hi All 

Is anyone else having trouble rendering the metadata and image for the Advanced collectible?

I have followed lesson 11 all the way to the end, everything works until I get to Opensea.  The NFT is shown in the Opensea Portal but none of the metadata nor is the image is shown. And yes, this is even after waiting 20 minutes.

To debug,  I added the little bit of else code below to the if statement in the set_tokenuri.py file  to check to see that if and what token uri was set:  

        # check to see if token uri has already been set
        if not advanced_collectible.tokenURI(token_id).startswith("https://"):
            print(f"Setting tokenURI of {token_id}")
            tokenURI = dog_metadata_dic[breed]
            set_tokenURI(token_id, advanced_collectible, tokenURI)
        # if the token uri is set print it
        else:
            print(
                f"{advanced_collectible.tokenURI(token_id)} is the token uri thats already set"
            )

Here's a print from running the set_tokenuri   script 5 times after creating  four x collectibles (3 Shibas and 1 St Bernard)
:

__Running 'scripts\advanced_collectible\set_tokenuri.py::main'...
Working on rinkeby
You have 4 tokenIds
Setting tokenURI of SHIBA_INU
https://ipfs.io/ipfs/QmdryoExpgEQQQgJPoruwGJyZmz6SqV4FRTX1i73CT3iXn?filename=1-SHIBA_INU.json is the token uri thats already set
Setting tokenURI of ST_BERNARD
https://ipfs.io/ipfs/QmbBnUjyHHN7Ytq9xDsYF9sucZdDJLRkWz7vnZfrjMXMxs?filename=2-ST_BERNARD.json 
is the token uri thats already set
Setting tokenURI of SHIBA_INU
https://ipfs.io/ipfs/QmdryoExpgEQQQgJPoruwGJyZmz6SqV4FRTX1i73CT3iXn?filename=1-SHIBA_INU.json is the token uri thats already set
Setting tokenURI of SHIBA_INU
https://ipfs.io/ipfs/QmdryoExpgEQQQgJPoruwGJyZmz6SqV4FRTX1i73CT3iXn?filename=1-SHIBA_INU.json is the token uri thats already set__

So the tokens are clearly being set. 

**The first Shiba-Iny has displayed correctly** (and this after about 60 attempts lol)
But then the following 3 NFTs don't.

What could be happening to stop the NFT metadata from rendering?

**Comment by Programmusic** [comment, marked as answer]
@PatrickAlphaC  do you have any thoughts about this? If my uri token is being set - why arent I getting any metadata shown?

**Reply by PatrickAlphaC** [reply]
Do you have a link to your smart contract address that you've deployed?

**Reply by Programmusic** [reply]
@PatrickAlphaC here is one I freshly baked today along with token uri:

0x9A0eB723c3acE4852884cbcA416BC677c0c6C398
ST BERNARD
https://testnets.opensea.io/assets/0x9A0eB723c3acE4852884cbcA416BC677c0c6C398/0
https://ipfs.io/ipfs/QmbBnUjyHHN7Ytq9xDsYF9sucZdDJLRkWz7vnZfrjMXMxs?filename=2-ST_BERNARD.json


I notice when I clicked the token URI its showing 2 x the json - like this

{
    "name": "ST_BERNARD",
    "description": "An adorable ST_BERNARD pup!",
    "image": "https://ipfs.io/ipfs/QmUPjADFGEKmfohdTaNcWhp7VGk26h5jXDA7v3VtTnTLcW?filename=st-bernard.png",
    "attributes": [
        {
            "trait_type": "cuteness",
            "value": 100
        }
    ]
}
{
"name": "ST_BERNARD",
"description": "An adorable ST_BERNARD pup!",
"image": "https://ipfs.io/ipfs/QmUPjADFGEKmfohdTaNcWhp7VGk26h5jXDA7v3VtTnTLcW?filename=st-bernard.png",
"attributes": [
{
"trait_type": "cuteness",
"value": 100
}
]
}
 is this relevant to my problem or just how my browser is handling the file?

and also while I am here, a **HUGE THANK YOU** for such an awesome tutorial video - you are a legend!

**Reply by PatrickAlphaC** [reply]
You just needed to hit the `refresh metadata` button and wait a few minutes.

https://testnets.opensea.io/assets/0x9A0eB723c3acE4852884cbcA416BC677c0c6C398/0

Great work, and glad you're enjoying!

<img width="738" alt="Screen Shot 2021-10-12 at 4 52 52 PM" src="https://user-images.githubusercontent.com/54278053/137027560-4bdae5e2-d2c0-41b8-a39c-ea54ee862c4c.png">
/0

**Reply by Programmusic** [reply]
🤣🤣🤣🤣🤣HAHAHAHHAHAHHA how funny ! 

I can't believe I was so dumb. **I was refreshing the chrome browser** instead of the metadata! 

I spent two days trying to figure this out, but at least I got to know the code in great detail and 
I  now have over  60 cute little doggies in my test account collection! 😊

Looking forward to completing the rest of the course now.

**Reply by PatrickAlphaC** [reply]
AWESOME WORK!!!

Sometimes, it's just something so simple :)
